modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/resource/OpenAPIResourceImpl.java

```diff
@@ -153,11 +152,13 @@ openAPI, _toOpenAPISpecFilter(mergedOpenAPISchemaFilter),
 		}
 
 		if (uriInfo != null) {
-			Server server = new Server();
+			for (String uriSchema : _URI_SCHEMES) {
```

Comment:
Hi @matijapetanjek! I see you're adding both http and https servers. I tested it locally and I see this:
<img width="394" alt="Screenshot 2022-09-16 at 11 11 52" src="https://user-images.githubusercontent.com/1937354/190602301-44d98c58-38d0-49fa-9e2d-8fa43d3a7220.png">
But my machine is not configured to use https, so I'd say the https server should not be there. Should we just return the server with the current protocol?

Reply:
@4lejandrito if I understood @dnebing correctly, on LR Cloud, client is accessing LR instance through a reverse proxy. Communication between client and reverse proxy is realized via https -> on reverse proxy https is terminated and -> communication between reverse proxy and LR instance is plain http protocol.

In this configuration Liferay is not configured with Https and when building server URLs inside liferay, we see only http protocol as enabled.(but for API Explorer URLs to work they need to use htpps) 
I see 2 solutions to this problem:

1. always generate http/https URLs and let user select the correct one(current implementation)
2. Maybe we can rely on X-Forwarded-Host and X-Forwarded-Proto headers to check if there is proxy in front. If those headers are missing, build URLs with enabled protocols from portal.properties? 

The reason I didn't go with 2nd option is because it's more complicated and I'm not sure that this is indeed the logic that would work in Cloud configuration. Maybe I can get back to Dave and ask him about it if you think that first approach is not good?

Reply:
But at the end, what we want to achieve is to respect whatever protocol is coming in the original request. So if we go to:

1. `http:domain/o/api`, the generated URLs and servers should be `http:domain/o/api`.
2. `https:domain/o/api`the generated URLs and servers should be `https:domain/o/api`.

I think it doesn't matter what happens in the middle (whether there is a reverse proxy or not), the expected end result is the same.

Reply:
API Explorer -> `https://reverseProxyHost/o/api` -> reverse proxy -> `http://liferayInstanceHost/o/api` -> Liferay 

Liferay gets `http://liferayInstanceDomain/o/api` in uriInfo, it doesn't know anything about `https://reverseProxyDomain/o/api` at least that's what I observed in debugger.
The way we're determining protocol now is